identity-iota/src/tangle/client.rs

```diff
@@ -97,6 +105,27 @@ impl Client {
       .map(|message| Receipt::new(self.network.clone(), message))
   }
 
+  /// Publishes arbitrary JSON data to the specified index on the Tangle.
+  /// Retries (promotes or reattaches) the message until it’s included (referenced by a milestone).
+  /// Default interval is 5 seconds and max attempts is 40.
+  pub async fn publish_json_with_retry<T: ToJson>(
+    &self,
+    index: &str,
+    data: &T,
+    interval: Option<u64>,
+    max_attempts: Option<u64>,
+  ) -> Result<Receipt> {
+    let receipt = self.publish_json(index, data).await?;
+    let reattached_messages = self
+      .client
+      .retry_until_included(receipt.message_id(), interval, max_attempts)
+      .await?;
```

Comment:
nit: very minor but the codebase convention is usually to annotate the type for new variables. Not sure what everyone's opinion on this is but it's the current convention.

E.g.
```rust
    let receipt: Receipt = self.publish_json(index, data).await?;
    let reattached_messages: Vec<MessageWrap (or whatever this returns)> = self
      .client
      .retry_until_included(receipt.message_id(), interval, max_attempts)
      .await?;
```

Reply:
I was not aware of this convention, but I kind of like it to be honest. I guess it makes it easier to figure things out when looking at code directly from GitHub.